### PR TITLE
MTL_1641 Patch Endpoint Update

### PR DESCRIPTION
**Pull request by evak6979**
## Link to JIRA ticket

https://hackney.atlassian.net/browse/MTTL-1641

## Describe this PR

Adding a Patch Endpoint

Add a summary of the problem and any additional context that helps explain what the issue we're trying to solve is.
#### _Checklist_

- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [x] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### CODE REVIEW

File: EqualityInformationApi/V1/Boundary/Request/Validation/EqualityInformationObjectValidator - Copy.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
+using FluentValidation;
+using Hackney.Core.Validation;
+using System;
+using System.Linq;
+
+namespace EqualityInformationApi.V1.Boundary.Request.Validation
+{
+    public class PatchEqualityInformationObjectValidator : AbstractValidator<PatchEqualityInformationObject>
```
Comment (amidomichael) [inline EqualityInformationApi/V1/Boundary/Request/Validation/EqualityInformationObjectValidator - Copy.cs:8]: the filename looks wrong for this?

File: EqualityInformationApi/V1/Gateways/IEqualityInformationGateway.cs [hunk 2]
```diff
@@ -11,5 +11,7 @@ namespace EqualityInformationApi.V1.Gateways
     public interface IEqualityInformationGateway
     {
         Task<EqualityInformation> Create(EqualityInformationObject request);
+
+        Task<EqualityInformation> Patch(PatchEqualityInformationObject request);
```
Comment (amidomichael) [inline EqualityInformationApi/V1/Gateways/IEqualityInformationGateway.cs:15]: I wouldn't use `Patch` internally in code, it's HTTP terminology

File: EqualityInformationApi/V1/UseCase/PatchUseCase.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            IEqualityInformationGateway gateway,
+            ISnsGateway snsGateway,
+            ISnsFactory snsFactory)
+        {
+            _gateway = gateway;
+            _snsGateway = snsGateway;
+            _snsFactory = snsFactory;
+        }
+
+        public async Task<EqualityInformationResponseObject> Execute(PatchEqualityInformationObject request, Token token)
```
Comment (amidomichael) [inline EqualityInformationApi/V1/UseCase/PatchUseCase.cs:29]: This method looks wrong - lots of references to equality information?

Comment (evak6979) [reply EqualityInformationApi/V1/UseCase/PatchUseCase.cs:29]: PatchEqualityInformation inherits from EqualityInformation. Save object save for Id.

File: EqualityInformationApi/V1/Gateways/IEqualityInformationGateway.cs (hunk 2 above)
Comment (evak6979) [reply EqualityInformationApi/V1/Gateways/IEqualityInformationGateway.cs:15]: Update?

File: EqualityInformationApi/V1/Boundary/Request/Validation/EqualityInformationObjectValidator - Copy.cs (hunk 1 above)
Comment (evak6979) [reply EqualityInformationApi/V1/Boundary/Request/Validation/EqualityInformationObjectValidator - Copy.cs:8]: Good spot

**Review by amidomichael — APPROVED** [review]

**Pull request merged**